segmentstore/server/src/main/java/io/pravega/segmentstore/server/writer/StorageWriter.java

```diff
@@ -594,11 +594,9 @@ public boolean isClosed() {
 
         @Override
         public long getLowestUncommittedSequenceNumber() {
-            // We collect the lowest value across all processors.
-            return this.processors.stream()
-                                  .mapToLong(WriterSegmentProcessor::getLowestUncommittedSequenceNumber)
-                                  .min()
-                                  .orElse(Operation.NO_SEQUENCE_NUMBER);
+            return this.processors.size() == 1
```

Comment:
In the docs for this method it says: "As opposed from {@link #getHighestCommittedSequenceNumber}, this method should be called for {@link WriterSegmentProcessor} instances that deal with the same Segment"

But both paths here are calling `getLowestUncommittedSequenceNumber`

Reply:
The Javadoc you are referring to is on the `AckCalculator` class. This is inside the `StorageWriter` class and is invoking that method correctly.

This is inside a private nested class named `ProcessorCollection` that holds all the processors for the same Segment. 

The difference here is if we only have one processor (every Stream Segment) then don't bother calling into `AckCalculator` and just return the result from its `SegmentAggregator`. For the (small minority of) TableProcessors, we go down the second path, which invokes the `AckCalculator`.

If you look inside the `AckCalculator.getLowestUncommitedSequenceNumber()` you'll see that you'll get the same result if you only pass in 1 item in the argument.

Reply:
For my own understanding, is it right that the check for single processor is only an optimization?  Would it be correct to replace:

```
this.processors.size() == 1
                    ? this.processors.get(0).getLowestUncommittedSequenceNumber()
                    : StorageWriter.this.ackCalculator.getLowestUncommittedSequenceNumber(this.processors);
```

with:

```
StorageWriter.this.ackCalculator.getLowestUncommittedSequenceNumber(this.processors);
```

Reply:
yes , they are equivalent. The vast majority of segments will be going down the first path.

Reply:
yes, they are equivalent. The vast majority of segments will go down the first path.